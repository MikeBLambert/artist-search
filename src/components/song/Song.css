.container {
  background-color: lightgrey;
  border: 5px solid black;
  box-shadow: 2px 2px grey;
  border-radius: 15px;
  margin-top: 50px;
  padding: 25px;
  text-align: center;
  font-family: 'Roboto', sans-serif;



  p {
    font-size: 20px;
  }
}