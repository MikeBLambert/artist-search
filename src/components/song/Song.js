import React, { Component, Fragment } from 'react';
import { getLyrics } from '../../services/lyricsApi';
import SongDetails from './SongDetails';

export default class Song extends Component {

  state = {
    lyrics: '',
    artist: '',
    song: '',
    loading: true,
  };

  componentDidMount() {
    const artist = this.props.location.pathname.slice(1).split('/')[1];
    const song = this.props.location.pathname.slice(1).split('/')[2];

    getLyrics(artist, song)
      .then(result => {
        this.setState({ lyrics: result.lyrics, artist, song, loading: false });
      });
  }

  render() {
    const { artist, song, lyrics, loading } = this.state;

    return (
      <Fragment>
        {!loading && <SongDetails artist={artist} song={song} lyrics={lyrics} />}
      </Fragment>
    );
  }
}
