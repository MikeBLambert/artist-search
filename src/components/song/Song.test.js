import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Song from './Song';
jest.mock('../../services/lyricsApi', () => {
  return { getLyrics: () => Promise.resolve({ lyrics: 'Lalalalala' }) };
});

describe('Song', () => {

  it('renders a search component', () => {
    const wrapper = shallow(<Song location={{ pathname: '/sldkfj' }}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a SongDetail component', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <Song />
      </MemoryRouter>
    );
    expect(wrapper.find('SongDetails')).toBeTruthy();
  });
});
