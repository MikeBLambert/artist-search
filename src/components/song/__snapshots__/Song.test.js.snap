// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song renders a search component 1`] = `
<Fragment>
  <div
    className="container"
  />
</Fragment>
`;
